### PR TITLE
Fix marking selected date in Datepicker of "DateField"

### LINKED ISSUE #312: DateField: selected date isn't marked as selected when using with "Form"
### Prerequisites

- [x] Can you reproduce the problem in a [MWE](https://en.wikipedia.org/wiki/Minimal_working_example)?
- [x] Are you running the latest version?
- [x] Did you check the FAQs to see if that helps you?
- [x] Are you reporting to the correct repository?
- [x] Did you perform a search in the issues?

### Description

Selected date isn't marked as selected in expanded Datapicker, if "DateFiled" is used inside a "Form".

### Steps to Reproduce

1. Go to https://precise-ui.io/#/Components/Form.

2. Use following code in section "View Code" of the first Example:

```javascript
const { Form, Button, TextField } = require('precise-ui');

<Form onSubmit={e => alert(JSON.stringify(e.data))}>
  <div>
    Name:
  </div>
  <div>
    <TextField name="name" />
  </div>
  <div>
    Date:
  </div>
  <div>
    <DateField name="someDate" />
  </div>
  <div>
    <Button>Submit</Button>
  </div>
</Form>
```

1a

3. Click on the field of the "DateField" and select new date from the Datapicker.

2a

3a

4. Click on the field of the "DateField".

**Expected behavior:**
Selected date is marked as selected in expanded Datapicker.

**Actual behavior:**
Selected date is not marked in expanded Datapicker.

4a

**Environment details:**
Browser: Chrome

### Possible Solution

It seems that the problme is in "DateFieldInt.part.tsx" in "change" function.
Solution proposal:

```javascript
private change = (date: Date | undefined, value: string) => {
    const { onChange, name = '', form } = this.props;

    if (!this.valueControlled) {
      this.setState({
        value,
        date,
      });
      if (form) {
        form.change({
          name,
          value,
        });
      } else {
        this.onOpenChange(true);
      }
    }

    if (typeof onChange === 'function') {
      onChange({
        value,
        date,
      });
    }
  };
```
PR with the solution proposal will be created within the next few days.

### PR DESCRIPTION
**Pull request by rad3k1**
### Prerequisites

Please make sure you can check the following two boxes:

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [x] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [ ] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

### Description

Fixes #312

### CODE REVIEW

File: src/components/DateField/DateFieldInt.part.tsx [hunk 1]
```diff
@@ -262,17 +262,16 @@ class DateFieldInt extends React.Component<DateFieldProps, DateFieldState> {
...
           value,
         });
       } else {
-        this.setState({
-          value,
-          date,
-        });
-
         this.onOpenChange(true);
       }
```
Comment (dimabory) [inline src/components/DateField/DateFieldInt.part.tsx:276]: could you consider this approach?

```tsx
if (!this.valueControlled) {
  this.setState({ value, date }, () => {
    form &&
      form.change({
        name,
        value,
      });
    this.onOpenChange(true);
  });
}
```

IMO: it looks much more readable...

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # Precise UI Changelog
 
+## 1.5.1
+- Fix marking selected date in Datepicker of `DateField`
```
Comment (dimabory) [inline CHANGELOG.md:4]:
```suggestion
## 1.5.1

- Fix marking selected date in Datepicker of `DateField`
```

File: src/components/DateField/DateFieldInt.part.tsx (hunk 1 above)
Comment (rad3k1) [reply src/components/DateField/DateFieldInt.part.tsx]: Done.

File: CHANGELOG.md (hunk 2 above)
Comment (rad3k1) [reply CHANGELOG.md]: Done.

**Review by dimabory — APPROVED** [review]
LGTM 🤝 Thanks @rad3k1

**Pull request merged**